plugins/AIAgents/Columns/AIAgentName.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/**
+ * Matomo - free/libre analytics platform
+ *
+ * @link    https://matomo.org
+ * @license https://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+declare(strict_types=1);
+
+namespace Piwik\Plugins\AIAgents\Columns;
+
+use Piwik\Plugin\Dimension\VisitDimension;
+use Piwik\Plugins\AIAgents\AIAgents;
+use Piwik\Tracker\Request;
+use Piwik\Tracker\Visitor;
+
+class AIAgentName extends VisitDimension
+{
+    protected $columnName   = 'ai_agent_name';
+    protected $columnType   = 'VARCHAR(40) NULL';
+    protected $nameSingular = 'AIAgents_AIAgentName';
+    protected $segmentName  = 'aiAgentName';
+    protected $type         = self::TYPE_TEXT;
+
+    public function __construct()
+    {
+        $this->suggestedValuesCallback = function () {
+            $values    = [];
+            $providers = AIAgents::getAvailableAgentProviders();
+
+            foreach ($providers as $provider) {
+                $values[] = $provider->getId();
+            }
+
+            return $values;
+        };
+    }
+
+    public function onNewVisit(Request $request, Visitor $visitor, $action)
+    {
+        $providers = AIAgents::getAvailableAgentProviders();
+
+        foreach ($providers as $provider) {
+            if ($provider->isDetectedForTrackerRequest($request)) {
+                return $provider->getId();
+            }
+        }
+
+        return false;
+    }
```

Comment:
Have you thought about how to handle the potential edge case, that a visit for whatever reason starts as a normal request, but one of the ongoing requests starts sending agent identifiers?
Currently I guess they  would be simply ignored. I was wondering if we should either update the current visit, by adding a `onExistingVisit` method, or implement a custom Tracker that enforces a new visit in case it changes.